### PR TITLE
Jobs APIupdate + departments grouping

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Pulls all jobs data from a new public `cowswap` specific Greenhouse endpoint
- Segments and displays jobs by departments (before we didn't have this data)
- Allows to filter jobs by a specific departments + shows the count

https://user-images.githubusercontent.com/31534717/162991628-1766e45f-079c-4f64-86ad-f1b38cb8d073.mov

No jobs:

Screen Shot 2022-04-12 at 15 52 03

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @fairlighteth , great changes!
However, seems that I have issues with the font in the dropdown: it looks different. 
font
font1

Besides, the dropdown itself is in white and looks different in comparison with your video above
in black

And a nitpick: could we please increase the margin between these lines? 
image

Could you please take a look into these issues?

Thanks!

File: pages/jobs.tsx [hunk 1]
```diff
@@ -1,29 +1,39 @@
...
-      dept.id === deptID ? jobsData.push(job) : null
-    }
-  })
+  const jobsData = {}
+  const { api } = siteConfig.greenhouse
+
+  try {
+    const response = await fetch(api)
+    const data = await response.json()
+    data.jobs.map(job => {
```
Comment (alfetopito) [inline pages/jobs.tsx:16]: You are not returning anything from the `map`, so a better function for this is `forEach`

**Comment by fairlighteth** [comment]
@elena-zh Addressed your comments. One thing that I haven't been able to change, is the grey hover background color for Windows. I'd suggest to leave as is for now.

Screen Shot 2022-04-13 at 11 47 54

**Review by alfetopito — APPROVED** [review]
Looks awesome!

Only one minor nitpick comment

**Review by elena-zh — APPROVED** [review]
@fairlighteth , looks great now!
Thank you!

File: pages/jobs.tsx (hunk 1 above)
Comment (fairlighteth) [reply pages/jobs.tsx]: Ah right, I'm not storing the return value of the `.map` in a variable. Updated and thanks for reviewing.

**Pull request merged**